File: java-version/src/main/java/bernard/tatin/procfs/LinuxProc.java
package bernard.tatin.procfs;

import java.nio.file.Path;
import java.nio.file.Paths;

class LinuxProc {
    private static final LinuxProc ourInstance = new LinuxProc();
    private final long pid = ProcessID.getPID();

    private Path innerProcPathName(String path) {
        return Paths.get("/proc/" + String.valueOf(pid) + "/" + path);
    }

    public static Path procPathName(String path) {
        return ourInstance.innerProcPathName(path);
    }

    private LinuxProc() {

    }
}

File: kotlin-version/src/main/java/bernard/tatin/ProcFS/LinuxProc.kt
package bernard.tatin.ProcFS

import java.nio.file.Path
import java.nio.file.Paths

internal class LinuxProc private constructor() {
    private val pid = ProcessID.getPID()

    private fun innerProcPathName(path: String): Path {
        return Paths.get("/proc/" + String.valueOf(pid) + "/" + path)
    }

    companion object {
        private val ourInstance = LinuxProc()

        fun procPathName(path: String): Path {
            return ourInstance.innerProcPathName(path)
        }
    }
}
